src/pages/api/snapshot/desktop/[name].ts

```diff
@@ -0,0 +1,23 @@
+// Next.js API route support: https://nextjs.org/docs/api-routes/introduction
+
+import type { NextApiRequest, NextApiResponse } from 'next'
+import { genScreenshot } from 'src/helpers'
+
+export interface IQuery {
+    url?: string
+    width?: number
+    height?: number
+}
+
+export default async (req: NextApiRequest, res: NextApiResponse) => {
+    const { url, width, height }= req.query as IQuery
```

Comment:
This is unsafe as query are always string. And that's why you transformed parameters in `genScreenshot`. It has to take responsible of its type here.